setup.py

```diff
@@ -18,6 +23,7 @@
     license="Apache-2.0",
     packages=find_packages(),
     install_requires=INSTALL_REQUIRES,
+    extras_require={"spark": ["pyspark>=2.2.0"]},
```

Comment:
This would essentially install the latest `pyspark` but that won't match with what the user potentially might have installed. I guess users might need to change this after the fact. Thoughts?

Reply:
Well, this says we need >= v2.2. If a user includes this in their requirements.txt with pyspark==2.4.0, I think our requirement won't prevent the user's installation from fetching the version of pyspark that they want, but this probably depends on their package management (i.e. I'm not 100% sure how pip, pipenv, conda, poetry would handle this situation).

Reply:
I agree. Some stuff will be up to end users to manage, and Spark is a bit weird as the pip version corresponds to the actual Spark version. LGTM. Approving.